src/ng/filter/orderBy.js

```diff
@@ -599,7 +602,7 @@ function orderByFilter($parse) {
         }
       }
 
-      return compare(v1.tieBreaker, v2.tieBreaker) * descending;
+      return (compare(v1.tieBreaker, v2.tieBreaker) || defaultCompare(v1.tieBreaker, v2.tieBreaker)) * descending;
```

Comment:
This is not intended to defer to the `defaultCompare` function over the user-provided function, but rather to provide a reasonable fallback should a user-provided function fail to the break the tie. See [#14481 (comment)](https://github.com/angular/angular.js/issues/14881#issuecomment-231164555)